include/flatbuffers/flatbuffers.h

```diff
@@ -363,9 +363,9 @@ template<typename T> class Vector {
   {
     return flatbuffers::span<T>(Data(), size());
```

Comment:
You can't use `Data()` here.
The library supports both little-endian and big-endian platforms. To get access to Vector's elements the `Get()` method should be used. For example, the Vector's iterator uses `Get()`.
There are only to cases when cast to a `span` is possible for `Vector<T>`:
- sizeof(T) == 1,
- T is `Offset<U>` (Vector of tables, structs, or strings).
all other cases are prohibited.

Reply:
We already have `data()` which returns a `T *`, which is unsafe in a similar way.

`Data()` has the comment `// The raw data in little endian format. Use with care.`, and `data()` should have that comment even more so (probably more explicitly, since it is more dangerous, as it promises a `T *` not just bytes.

Getting a `span` is not any different from `data()`, so I'd be fine it being allowed, as long as it also has a strongly worded comment about endianness. Maybe bot `data()` and this method should have a `FLATBUFFERS_ASSERT(FLATBUFFERS_LITTLEENDIAN)` or similar inside them.

Having these functions is still useful, most software today does never runs on big endian machines anymore so this kind of access is usually ok.

Reply:
Personally, I avoid using a platform-dependent interface only implementations can be platform dependent.

About this proposal,

The proposed implementation of it as a platform-dependent **implicit** cast operator looks unsafe.
What reason to make this method a member function (**implicit** cast operator) but not as a non-member free-standing method like `MakeAsSpan<VT>(VT &v)`?

Compatibility checking of `FLATBUFFERS_LITTLEENDIAN` should be done at compile time.

Reply:
@OptoCloud could you add a compile-time guard method to check that a `T` is compatible with the raw `T[N]` view?
What do you think about a non-member conversion method?